modules/dataframe/include/inviwo/dataframe/util/dataframeutil.h

```diff
@@ -109,8 +111,66 @@ std::shared_ptr<DataFrame> IVW_MODULE_DATAFRAME_API
 combineDataFrames(std::vector<std::shared_ptr<DataFrame>> dataframes, bool skipIndexColumn = false,
                   std::string skipcol = "index");
 
+/**
+ * \brief apply predicate \pred to each value of column \col and return the row indices where the
+ * predicate evaluates to true.
+ *
+ * Note: the predicate function needs to take care of the different column datatypes
+ * \code{.cpp}
+ * auto pred = [](const auto& arg) { 
+ *                 if  constexpr(std::is_same_v<decltype(arg), float>) { 
+ *                     return  arg == 4.5f
+ *                 } else {
+ *                     return true;
+ *                 }};
+ * \endcode
+ *
+ * Alternatively, predicate overloads can be used.
+ * \code{.cpp}
+ * #include <inviwo/core/util/stdextensions.h>
+ *
+ * auto pred = util::overloaded{[](const std::string& arg) { return arg == "bla"; }, 
+ *                              [](const int& arg) { return arg == 5;  }, 
+ *                              [](const auto&) { return true; });
+ * \endcode
+ *
+ * @param col   column containing data for filtering
+ * @param pred  predicate to check values from \p col
+ * @return list of row indices where rows fulfill the predicate
+ */
+template <typename Pred>
+std::vector<size_t> filteredRows(std::shared_ptr<const Column> col, Pred pred);
```

Comment:
Can usage be simplified if the column type is somehow specified? E.g. passing the filter along to the dispatching?
And/or might be better to have an example that checks for floating point/integral numbers.

Reply:
The main problem here is that you don't know the type of your column. Hence, the predicate needs to handle all the different types, mainly distinguish between strings/categorical and numerical values.
If you know the column type, then one could potentially have a templated function based on the type (which is needed for the predicate). 
Making such a function a member function of Column does not work as it cannot be virtual due to the template.